job-runner/chain/traverser.go

```diff
@@ -377,3 +386,36 @@ func (t *traverser) sendJL(pJob proto.Job, err error) {
 		jLogger.Errorf("problem sending job log (%#v) to the RM: %s", jl, err)
 	}
 }
+
+func (t *traverser) retrySequence(j proto.Job) {
```

Comment:
Please avoid `j` because it's nearly impossible to follow or find in code. You'll notice `pJob` and `doneJ` (which should probably be `doneJob`). There's also `realJob` in  other code. These names are more illuminating and far easier to find/follow.

Reply:
ok, assuming `pJob` is short of `protoJob` and not `pendingJob`...I did `s/j/pJob` for this. I also did `s/doneJ/doneJob`.

Reply:
actually i'll rename `j` to `failedJob`